apcd-cms/src/apps/utils/apcd_database.py

```diff
@@ -1104,6 +1104,73 @@ def create_extension(form, iteration, sub_data):
         if conn is not None:
             conn.close()
 
+def update_extension(form):
+    cur = None
+    conn = None
+    try:
+        conn = psycopg2.connect(
+            host=APCD_DB['host'],
+            dbname=APCD_DB['database'],
+            user=APCD_DB['user'],
+            password=APCD_DB['password'],
+            port=APCD_DB['port'],
+            sslmode='require'
+        )
+        cur = conn.cursor()
+        operation = """UPDATE extensions
+            SET
+            updated_at= %s,"""
+        
+        set_values = []
+        # to set column names for query to the correct DB name
+        columns = {
+            'applicable-data-period': 'applicable_data_period',
+            'status': 'status',
+            'outcome': 'outcome',
+            'approved': 'approved_expiration_date',
+            'notes': 'notes'
+        }
+        # To make sure fields are not blank. 
+        # If they aren't, add column to update operation
+        for field, column_name in columns.items():
+            value = form.get(field)
+            if value not in (None, ""):
+                set_values.append(f"{column_name} = %s")
+
+        operation += ", ".join(set_values) + " WHERE extension_id = %s"
+        ## add last update to all extension updates
+        values = [
+            datetime.now(),
+        ]
+        
+        for field, column_name in columns.items():
+            value = form.get(field)
+            if value not in (None, ""):
+                # to make sure applicable data period field is the right type for insert to DB
+                if column_name == 'applicable_data_period':
+                    values.append(int(value.replace('-', '')))
+        # server side clean values
+                else:
+                    values.append(_clean_value(value))
+        ## to make sure extension id is last in query to match with WHERE statement
+        values.append(_clean_value(form['extension_id']))
+
+        cur.execute(operation, values)
+        print("operation =")
+        print(operation)
+        print("values = ")
+        print(values)
+        print("success")
```

Comment:
nit: debug lines will be removed before final commit, right?

Reply:
Was cleaning some stuff up before review. Should be good to review now! Thank you